kalite/topic_tools/annotate.py

```diff
@@ -50,97 +50,71 @@ def update_content_availability(content_list, language="en"):
+                update["available"] = True
 
         elif content.get("kind") == "Topic":
             # Ignore topics, as we only want to update their availability after we have updated the rest.
             continue
         else:
+            file_id = content.get("youtube_id", content.get("id"))
             default_thumbnail = create_thumbnail_url(content.get("id"))
-            dubmap = i18n.get_id2oklang_map(content.get("id"))
-            if dubmap:
-                content_lang = i18n.select_best_available_language(language, available_codes=dubmap.keys()) or ""
-                if content_lang:
-                    dubbed_id = dubmap.get(content_lang)
-                    format = content.get("format", "")
-                    if (dubbed_id + "." + format) in contents_folder:
-                        update["available"] = True
-                        thumbnail = create_thumbnail_url(dubbed_id) or default_thumbnail
-                        update["content_urls"] = {
-                            "stream": django_settings.CONTENT_URL + dubmap.get(content_lang) + "." + format,
-                            "stream_type": "{kind}/{format}".format(kind=content.get("kind").lower(), format=format),
-                            "thumbnail": thumbnail,
-                        }
-                    elif django_settings.BACKUP_VIDEO_SOURCE:
-                        update["available"] = True
-                        update["content_urls"] = {
-                            "stream": django_settings.BACKUP_VIDEO_SOURCE.format(youtube_id=dubbed_id, video_format=format),
-                            "stream_type": "{kind}/{format}".format(kind=content.get("kind").lower(), format=format),
-                            "thumbnail": django_settings.BACKUP_VIDEO_SOURCE.format(youtube_id=dubbed_id, video_format="png"),
-                        }
+            format = content.get("format", "")
+            filename = file_id + "." + format
 
             # Get list of subtitle language codes currently available
             subtitle_lang_codes = subtitle_langs.get("{id}.srt".format(id=content.get("id")), [])
 
-            # Generate subtitle URLs for any subtitles that do exist for this content item
-            subtitle_urls = [{
-                "code": lc,
-                "url": django_settings.STATIC_URL + "srt/{code}/subtitles/{id}.srt".format(code=lc, id=content.get("id")),
-                "name": i18n.get_language_name(lc)
-                } for lc in subtitle_lang_codes]
-
-            # Sort all subtitle URLs by language code
-            update["subtitle_urls"] = sorted(subtitle_urls, key=lambda x: x.get("code", ""))
-
-        with i18n.translate_block(language):
-            update["title"] = _(content.get("title"))
-            update["description"] = _(content.get("description")) if content.get("description") else ""
-
+            if filename in contents_folder or language in subtitle_lang_codes:
```

Comment:
Fallback to using a subtitled video with a different language video if available.